lms/djangoapps/courseware/views/views.py

```diff
@@ -737,6 +737,13 @@ def _progress(request, course_key, student_id):
         and certs_api.cert_generation_enabled(course_key)
     )
 
+    # If current certificate is invalidated by instructor
+    # then don't show the generate button.
+    is_invalid = False
+    if show_generate_cert_btn and certs_api.is_certificate_invalid(student, course_key):
```

Comment:
Above code increases the query count. You can see here.
https://build.testeng.edx.org/job/edx-platform-python-unittests-pr/20273/#showFailuresLink

Because this method `certs_api.is_certificate_invalid` called each time when page is loaded whether courses has certificate configuration or not. 
With following code this method is called only when course has active config.
```
certificate_invalidated = False
if show_generate_cert_btn and certs_api.is_certificate_invalid(student, course_key):
        show_generate_cert_btn = False
        certificate_invalidated = True
```